gitbutler-ui/src/lib/components/CommitDialog.svelte

```diff
@@ -39,31 +40,55 @@
 	export let cloud: ReturnType<typeof getCloudApiClient>;
 	export let user: User | undefined;
 	export let selectedOwnership: Writable<Ownership>;
+	export const expanded = persisted<boolean>(false, 'commitBoxExpanded_' + branch.id);
 
 	const aiGenEnabled = projectAiGenEnabled(projectId);
 	const runCommitHooks = projectRunCommitHooks(projectId);
 	const currentCommitMessage = projectCurrentCommitMessage(projectId, branch.id);
-	export const expanded = persisted<boolean>(false, 'commitBoxExpanded_' + branch.id);
 
-	let commitMessage: string = get(currentCommitMessage) || '';
-	let isCommitting = false;
-	let textareaElement: HTMLTextAreaElement;
-	$: if (textareaElement && commitMessage && expanded) {
-		textareaElement.style.height = 'auto';
-		textareaElement.style.height = `${textareaElement.scrollHeight + 2}px`;
+	function getCommitMessageTitleAndDescription(commitMessage: string) {
+		// Split the commit message into title and description
+		// get the first line as title and the rest as description
+		const [summary, ...description] = commitMessage.split('\n');
+		return {
+			summary: summary || '',
+			// remove top new line
+			description: description.join('\n').replace(/^\n/, '') || ''
+		};
+	}
+
+	function concatCommitMessage(message: { summary: string; description: string }) {
+		return `${message.summary}\n${message.description}`;
 	}
 
+	let commitMessageSet = getCommitMessageTitleAndDescription(get(currentCommitMessage));
```

Comment:
You should be able to replace `get(currentCommitMessage)` with `$currentCommitMessage`, but I would also see if we can omit having a `commitMessageSet` type object and instead destructure this declaration into commitTitle and commitDescription.

Reply:
Got it